app/models.py

```diff
@@ -388,6 +391,17 @@ def serialize(self, data=None):
             "countersignedAt": countersigned_at,
         }, **(data or {}))
 
+        if self.agreement_details and self.agreement_details.get('uploaderUserId'):
+            user = User.query.filter(
+                User.id == self.agreement_details.get('uploaderUserId')
+            ).first()
+
+            if user:
```

Comment:
Should this fail horribly if there's no user found with the user ID supplied?  Feels like that should be an error - we don't want to store "uploaderUserId" if it isn't associated with a user account.